lib/stdio/encoding.js

```diff
@@ -12,14 +12,21 @@ export const handleStreamsEncoding = (stdioStreams, {encoding}, isSync) => {
 	}
 
 	const objectMode = newStdioStreams.findLast(({type}) => type === 'generator')?.value.objectMode === true;
-	const encodingEndGenerator = objectMode ? encodingEndObjectGenerator : encodingEndStringGenerator;
-	const transform = encodingEndGenerator.bind(undefined, encoding);
+	const stringDecoder = new StringDecoder(encoding);
+	const generator = objectMode
+		? {
+			transform: encodingEndObjectGenerator.bind(undefined, stringDecoder),
+		}
+		: {
+			transform: encodingEndStringGenerator.bind(undefined, stringDecoder),
+			final: encodingEndStringFinal.bind(undefined, stringDecoder),
+		};
```

Comment:
We use a few transforms internally. This PR rewrites them to using the new syntax.